admin/class-perfecty-push-admin.php

```diff
@@ -578,11 +583,12 @@ public function print_send_notification_page() {
 				$item['perfecty-push-send-notification-message']     = sanitize_textarea_field( $item['perfecty-push-send-notification-message'] );
 				$item['perfecty-push-send-notification-url-to-open'] = sanitize_text_field( $item['perfecty-push-send-notification-url-to-open'] );
 				$item['perfecty-push-send-notification-image']       = sanitize_text_field( $item['perfecty-push-send-notification-image'] );
+				$item['perfecty-push-send-notification-timeoffset']  = sanitize_text_field( $item['perfecty-push-send-notification-timeoffset'] );
 
-				$payload = Perfecty_Push_Lib_Payload::build( $item['perfecty-push-send-notification-message'], $item['perfecty-push-send-notification-title'], $item['perfecty-push-send-notification-image'], $item['perfecty-push-send-notification-url-to-open'] );
-
+				$payload    = Perfecty_Push_Lib_Payload::build( $item['perfecty-push-send-notification-message'], $item['perfecty-push-send-notification-title'], $item['perfecty-push-send-notification-image'], $item['perfecty-push-send-notification-url-to-open'] );
+				$timeoffset = intval( $item['perfecty-push-send-notification-timeoffset'] );
```

Comment:
Instead of the timeoffset we could receive the date in string format and convert it to DateTime to do the timeoffset calculation in the PHP code, check my other comment in `lib/class-perfecty-push-lib-push-server.php`